files/build_templates/docker_image_ctl.j2

```diff
@@ -151,11 +151,11 @@ function waitForAllInstanceDatabaseConfigJsonFilesReady()
                     fi
 		done
             done
-	fi
```

Comment:
Since we are moving the waitForAllInstanceDatabaseConfigJsonFilesReady call before the check 
```
until [[ ($(docker exec -i database$DEV pgrep -x -c supervisord) -gt 0) && ($($SONIC_DB_CLI PING | grep -c PONG) -gt 0) &&
```
can we change the if [ ! -z "$DEV" ]; then in **line 133** to if [ -z "$DEV" ]; then ?

Also do we really need to add sleep 1 delay here

Reply:
@mlok-nokia could you check this again

Reply:
> Since we are moving the waitForAllInstanceDatabaseConfigJsonFilesReady call before the check
> 
> ```
> until [[ ($(docker exec -i database$DEV pgrep -x -c supervisord) -gt 0) && ($($SONIC_DB_CLI PING | grep -c PONG) -gt 0) &&
> ```
> 
> can we change the if [ ! -z "$DEV" ]; then in **line 133** to if [ -z "$DEV" ]; then ?
> 
> Also do we really need to add sleep 1 delay here

We cannot change the if { ! -z "$DEV" ] to if [ -z "$DEV" ].   The negative condition check is the main key for making sure the instance database# start need to wait for the Redis reday.   Although  instance database# depends on the host database. But when database is started, the instance database# will be started right after it.   This is the reason which we need to use this waitForAllInstanceDatabaseConfigJsonFilesReady  function,    

I think we can remove 1 second delay. which is at the end.  Let me verify it and I will update the PR